spark/src/main/java/org/apache/iceberg/spark/SparkTableUtil.java

```diff
@@ -0,0 +1,682 @@
+import org.apache.parquet.hadoop.ParquetFileReader;
+import org.apache.spark.TaskContext;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.api.java.function.FlatMapFunction;
+import org.apache.spark.api.java.function.MapFunction;
+import org.apache.spark.api.java.function.MapPartitionsFunction;
+import org.apache.spark.sql.AnalysisException;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Encoders;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.catalyst.TableIdentifier;
+import org.apache.spark.sql.catalyst.analysis.NoSuchDatabaseException;
+import org.apache.spark.sql.catalyst.analysis.NoSuchTableException;
+import org.apache.spark.sql.catalyst.analysis.UnresolvedAttribute;
+import org.apache.spark.sql.catalyst.catalog.CatalogTable;
+import org.apache.spark.sql.catalyst.catalog.CatalogTablePartition;
+import org.apache.spark.sql.catalyst.catalog.SessionCatalog;
+import org.apache.spark.sql.catalyst.expressions.Expression;
+import org.apache.spark.sql.catalyst.expressions.NamedExpression;
+import org.apache.spark.sql.catalyst.parser.ParseException;
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan;
+import scala.Function2;
+import scala.Option;
+import scala.Some;
+import scala.Tuple2;
+import scala.collection.JavaConverters;
+import scala.runtime.AbstractPartialFunction;
+
+import static org.apache.spark.sql.functions.col;
+
+/**
+ * Java version of the original SparkTableUtil.scala
+ * https://github.com/apache/iceberg/blob/apache-iceberg-0.8.0-incubating/spark/src/main/scala/org/apache/iceberg/spark/SparkTableUtil.scala
+ */
+public class SparkTableUtil {
+
+  private static final PathFilter HIDDEN_PATH_FILTER =
+      p -> !p.getName().startsWith("_") && !p.getName().startsWith(".");
+
+  private SparkTableUtil() {
+  }
+
+  /**
+   * Returns a DataFrame with a row for each partition in the table.
+   *
+   * The DataFrame has 3 columns, partition key (a=1/b=2), partition location, and format
+   * (avro or parquet).
+   *
+   * @param spark a Spark session
+   * @param table a table name and (optional) database
+   * @return a DataFrame of the table's partitions
+   */
+  public static Dataset<Row> partitionDF(SparkSession spark, String table) {
+
+    List<SparkPartition> partitions = getPartitions(spark, table);
+    return spark.createDataFrame(partitions, SparkPartition.class).toDF("partition", "uri", "format");
+  }
+
+  /**
+   * Returns a DataFrame with a row for each partition that matches the specified 'expression'.
+   *
+   * @param spark      a Spark session.
+   * @param table      name of the table.
+   * @param expression The expression whose matching partitions are returned.
+   * @return a DataFrame of the table partitions.
+   */
+  public static Dataset<Row> partitionDFByFilter(
+      SparkSession spark,
+      String table,
+      String expression) {
```

Comment:
Style: could you leave arguments on the same line?

If the line is too long, you can either wrap and indent to the start of the argument list, or start the entire argument list at 2 indentations (4 spaces) on the next line. Either one of these:

```java
  public void method(Arg arg1,
                     Arg arg2) { ... }

  public void method(
      Arg arg1, Arg arg2) { ... }
```